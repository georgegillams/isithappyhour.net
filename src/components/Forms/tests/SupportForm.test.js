import React from 'react';
import { render } from '@testing-library/react';

import { SupportForm } from '..';

describe('<SupportForm />', () => {
  it('Should render correctly', () => {
    const { container } = render(
      <SupportForm link={{ id: 'some-link', name: 'name 1' }} onDataChanged={() => null} onSubmit={() => null} />
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with no link id', () => {
    const { container } = render(<SupportForm link={{}} onDataChanged={() => null} onSubmit={() => null} />);

    expect(container).toMatchSnapshot();
  });
});
