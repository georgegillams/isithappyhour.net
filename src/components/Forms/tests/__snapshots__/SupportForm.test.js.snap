// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SupportForm /> Should render correctly 1`] = `
<div>
  <div
    class="sc-crXcEl"
  >
    <label
      class="sc-fEOsli iHSlJz"
      for="description_FORM_ID"
    >
      Description
    </label>
    <div
      class="sc-iBkjds hDuMmX sc-breuTD bOZwsL"
      placeholder="Description"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        class="sc-dIouRR cqKYoJ"
        id="description_FORM_ID"
        name="Description"
        value=""
      />
      <svg
        class="sc-gKXOVf ktoyia sc-hHLeRK kBmlVx"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <label
      class="sc-fEOsli iHSlJz"
      for="url_FORM_ID"
    >
      URL
    </label>
    <div
      class="sc-iBkjds hDuMmX sc-breuTD bOZwsL"
      placeholder="URL"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        class="sc-dIouRR cqKYoJ"
        id="url_FORM_ID"
        name="URL"
        value=""
      />
      <svg
        class="sc-gKXOVf ktoyia sc-hHLeRK kBmlVx"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <button
      class="sc-dkzDqf bOfPfm sc-evZas hAEDCd"
      type="button"
    >
      <span
        class="sc-eCYdqJ ghZAyu"
      >
        Add link
      </span>
    </button>
  </div>
</div>
`;

exports[`<SupportForm /> Should render correctly with no link id 1`] = `
<div>
  <div
    class="sc-crXcEl"
  >
    <label
      class="sc-fEOsli iHSlJz"
      for="name_FORM_ID"
    >
      Name
    </label>
    <div
      class="sc-iBkjds hDuMmX sc-breuTD bOZwsL"
      placeholder="Name"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        class="sc-dIouRR cqKYoJ"
        id="name_FORM_ID"
        name="Name"
        value=""
      />
      <svg
        class="sc-gKXOVf ktoyia sc-hHLeRK kBmlVx"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <label
      class="sc-fEOsli iHSlJz"
      for="description_FORM_ID"
    >
      Description
    </label>
    <div
      class="sc-iBkjds hDuMmX sc-breuTD bOZwsL"
      placeholder="Description"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        class="sc-dIouRR cqKYoJ"
        id="description_FORM_ID"
        name="Description"
        value=""
      />
      <svg
        class="sc-gKXOVf ktoyia sc-hHLeRK kBmlVx"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <label
      class="sc-fEOsli iHSlJz"
      for="url_FORM_ID"
    >
      URL
    </label>
    <div
      class="sc-iBkjds hDuMmX sc-breuTD bOZwsL"
      placeholder="URL"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        class="sc-dIouRR cqKYoJ"
        id="url_FORM_ID"
        name="URL"
        value=""
      />
      <svg
        class="sc-gKXOVf ktoyia sc-hHLeRK kBmlVx"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <button
      class="sc-dkzDqf bOfPfm sc-evZas hAEDCd"
      type="button"
    >
      <span
        class="sc-eCYdqJ ghZAyu"
      >
        Add link
      </span>
    </button>
  </div>
</div>
`;
